### PR TITLE
chore(manage): added doc for serving signoz from an external url / prefix

### PR DESCRIPTION
**Pull request by SagarRajput-7**
## Summary

Related - https://github.com/SigNoz/platform-pod/issues/1775

## Motivation / Problem

Doc for New feature - Serving Signoz from a prefix

## Changes
- Added new doc page under `manage` 

## Description

Doc for New feature - Serving Signoz from a prefix

## Type of Change

- [x] **Docs** – Changes to `data/docs/**`
- [ ] **Blog** – Changes to `data/blog/**`
- [ ] **Site Code** – Changes to `app/**`, `components/**`, `hooks/**`, `utils/**`, config, etc.
- [ ] **Redirects** – Renamed/moved docs or updated `next.config.js` redirects
- [ ] **Dependencies / CI / Scripts**
- [ ] **Other** – Explain below

## Impact

- [x] Documentation only
- [ ] UI changes
- [ ] Developer workflow
- [ ] Performance impact
- [ ] Breaking change

If breaking change, describe migration steps:

### General
- [x] Branch is up to date with `main`
- [x] PR title follows conventional format (`type: description`)
- [x] Self-reviewed the code
- [ ] Comments added for complex logic

### For all changes
- [x] Built locally (`yarn build`) with no errors
- [ ] Ran `yarn lint` and fixed any issues
- [ ] Pre-commit hooks passed (or ran `yarn check:doc-redirects` / `yarn check:docs-metadata` if applicable)

### For docs changes (`data/docs/**`)
- [x] Followed the docs author checklist in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [x] Added/updated the page in `constants/docsSideNav.ts` if adding or moving a doc
- [ ] Any added docs images use WebP format and live under `public/img/docs/<topic>/`

### For blog changes
- [ ] Followed the blog workflow in [contributing/blog-workflow.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/blog-workflow.md)
- [ ] Frontmatter includes `title`, `date`, `author`, `tags` (and `canonicalUrl` if applicable)
- [ ] Images use WebP format and live under `public/img/blog/<YYYY-MM>/`

### For site code changes
- [ ] Followed the site code playbook in [contributing/site-code.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/site-code.md)
- [ ] New dependencies are justified in the PR description (if any)

### For renamed or moved docs
- [ ] Followed the redirects and discovery section in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added permanent redirect in `next.config.js` under `async redirects()`
- [ ] Updated internal links and sidebar in `constants/docsSideNav.ts`
- [ ] Ran `yarn check:doc-redirects` to verify

## Testing

- [x] Tested locally
- [ ] Edge cases considered
- [ ] Existing functionality verified
- [ ] New tests added (if applicable)

Steps to test:
1.
2.
3.

## Additional Context

---

**Note:** Submit as Draft by default. Mark "Ready for review" when checks pass and content is ready.

### CODE REVIEW

File: data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+date: 2026-04-30
+id: serving-on-external-url
+title: Serving SigNoz from a Prefix
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx]: done

File: data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+To serve SigNoz under a URL prefix, set environment variable `SIGNOZ_GLOBAL_EXTERNAL__URL` on the `signoz` service to the full external URL, including the path:
+
+```bash
+SIGNOZ_GLOBAL_EXTERNAL__URL=https://example.com/signoz
+```
+
+SigNoz reads the path component (`/signoz`) at startup and applies it as the prefix for all API routes and frontend assets.
+
+
+## What SigNoz handles automatically
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx]: done

File: data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Health endpoints (`/api/v1/health`, `/api/v2/healthz`, `/api/v2/readyz`, `/api/v2/livez`) are served **without** the prefix so existing container probes keep working without reconfiguration.
+</Admonition>
+
+## Validate
+
+After restarting SigNoz with the updated configuration:
+
+1. Open `https://example.com/signoz/` in your browser.
+2. Confirm the SigNoz UI loads and in-app navigation works end to end.
+3. Verify health probes succeed - confirm `GET /api/v1/health` (without the prefix) returns `200 OK`.
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx]: done the curl part, other is not needed

File: constants/docsSideNav.ts [hunk 4]
```diff
@@ -518,6 +518,12 @@ const docsSideNav = [
                 label: 'Relational Databases Support',
                 isExpanded: false,
               },
+              {
+                type: 'doc',
+                route: '/docs/manage/administrator-guide/configuration/serving-on-external-url',
+                label: 'Serving from a Prefix',
+                isExpanded: false,
+              },
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply constants/docsSideNav.ts:525]: done

File: data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Configuration
+
+To serve SigNoz under a URL prefix, set environment variable `SIGNOZ_GLOBAL_EXTERNAL__URL` on the `signoz` service to the full external URL, including the path:
+
+```bash
+SIGNOZ_GLOBAL_EXTERNAL__URL=https://example.com/signoz
+```
+
+SigNoz reads the path component (`/signoz`) at startup and applies it as the prefix for all API routes and frontend assets.
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx]: not needed, information overload there are too many ways to cover to configure signoz, also the parent doc covers the configuration (popular ones)

File: data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+This feature is available from SigNoz v0.121.0 or later.
+</Admonition>
+
+## Configuration
+
+Configure the base path by setting `SIGNOZ_GLOBAL_EXTERNAL__URL` to your external URL:
+
+```bash
+SIGNOZ_GLOBAL_EXTERNAL__URL=https://example.com/signoz
+```
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx:24]: not needed, information overload. There are too many ways to cover to configure signoz, also the parent doc covers the configuration (popular ones)

File: data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+SigNoz reads the path component from the URL (e.g. `/signoz`) and uses it as the base path for all API routes and frontend assets. Health endpoints (`/api/v1/health`, `/api/v2/healthz`, `/api/v2/readyz`, `/api/v2/livez`) remain accessible without the prefix for container healthchecks.
+</Admonition>
+
+## Validate
+
+After restarting SigNoz with the updated configuration:
+
+1. Open `https://example.com/signoz/` in your browser.
+2. Confirm the SigNoz UI loads and in-app navigation works end to end.
+3. Verify health probes succeed - run the following and confirm it returns `200 OK`:
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx:38]: Not needed, it would be specific, users should check if the app is running in general

File: data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+tags: [Self-Host]
+---
+
+SigNoz can be served under a base path, so you can place it behind a reverse proxy alongside other services on the same domain - for example, `https://example.com/signoz/`.
+
+A single environment variable is the only configuration required. Everything under SigNoz derives the base path from its value at startup. No image rebuild is needed.
+
+<Admonition type="info">
+This feature is available from SigNoz v0.121.0 or later.
+</Admonition>
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply data/docs/manage/administrator-guide/configuration/serving-on-external-url.mdx:16]: already have a callout for this

**Review by Nageshbansal — APPROVED** [review]

**Review by therealpandey — APPROVED** [review]
